Subject: [PATCH] ohci1394: log physical posted write errors
From: Stefan Richter <stefanr@s5r6.in-berlin.de>
Date: 1138684429 -0500

Give a warning if incoming write requests to the physical range fail.

Signed-off-by: Stefan Richter <stefanr@s5r6.in-berlin.de>
Signed-off-by: Jody McIntyre <scjody@modernduck.com>
Signed-off-by: Olaf Hering <olh@suse.de>

---

 drivers/ieee1394/ohci1394.c |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

1be7be03e875c53ac7575ab458606dc14ae199e5
diff --git a/drivers/ieee1394/ohci1394.c b/drivers/ieee1394/ohci1394.c
index b6b96fa..eb329a8 100644
--- a/drivers/ieee1394/ohci1394.c
+++ b/drivers/ieee1394/ohci1394.c
@@ -572,6 +572,7 @@ static void ohci_initialize(struct ti_oh
 		  OHCI1394_reqTxComplete |
 		  OHCI1394_isochRx |
 		  OHCI1394_isochTx |
+		  OHCI1394_postedWriteErr |
 		  OHCI1394_cycleInconsistent);
 
 	/* Enable link */
@@ -2374,7 +2375,10 @@ static irqreturn_t ohci_irq_handler(int 
 
 		event &= ~OHCI1394_unrecoverableError;
 	}
-
+	if (event & OHCI1394_postedWriteErr) {
+		PRINT(KERN_ERR, "physical posted write error");
+		/* no recovery strategy yet, had to involve protocol drivers */
+	}
 	if (event & OHCI1394_cycleInconsistent) {
 		/* We subscribe to the cycleInconsistent event only to
 		 * clear the corresponding event bit... otherwise,
